downstreamadapter/dispatchermanager/event_dispatcher_manager.go

```diff
@@ -204,6 +215,27 @@ func NewEventDispatcherManager(
 		if err != nil {
 			return nil, 0, errors.Trace(err)
 		}
+		// redo meta manager
+		if manager.redoManager.Enabled() {
```

Comment:
Could we move these logic into L260, to make all redo related logic together? Maybe they can be a seperate function to be more clear.